lib/transactions-common.js

```diff
@@ -1426,8 +1426,10 @@ Transact.prototype._processTransaction = function (txid, description, items, con
   
   // First, need to iterate over the changes that are going to be made and make sure that,
   // if there are hard removes, the db version of the doc gets stored on the transaction
+  // We only do this step if the remove is not instant - otherwise the doc would already have
+  // been removed from the collection by this point.   
   _.each(items, function (item, index) {
-    if (item.action === "remove" && item.hardDelete) {
+    if (item.action === "remove" && item.hardDelete && !item.instant) {
       // Get the existing doc and store it in the transaction record
       // We overwrite the temporary version of the doc from an instant remove on the client
       // Because chances are that the whole document was not available on the client
```

Comment:
Instant removes happens in `Transact.prototype._doRemove`

The method is called at the time you add the remove action to the transaction. 

in this block:

    if (instant) {
      var fullDoc = collection.findOne(sel); // , {transform: null}
      fields.doc = fullDoc;
      if (!fullDoc) {
        // There is no existing document to remove
        // Return without creating an item in the queue or recording it
        self.log("Document not found for removal:", sel);
      }
      else {
        fields.doc.transaction_id = self._transaction_id;
        fields.doc.deleted = ServerTime.date();
      }
      var item = self._createItem("remove", collection, _id, fields, instant, self._permissionCheckOverridden(opt));
      self._recordTransaction(item);
      collection.remove(sel, callback);
    }


So if it's an instant delete, we should already have the doc saved on the txn item by now. In fact - by now it's too late to add the doc since it's been deleted. 

I'm not sure exactly the use case the comment is referring to.